storage/test/test_cloud_storage.py

```diff
@@ -315,3 +318,28 @@ def test_rechunked(max_size, parts):
     assert len(rechunked) == len(parts)
     for index, chunk in enumerate(rechunked):
         assert chunk == parts[index]
+
+
+@pytest.mark.parametrize("path", ["/", _TEST_PATH])
+def test_clean_partial_uploads(storage_engine, path):
+
+    # Setup root path and add come content to _root_path/uploads
+    storage_engine._root_path = path
+    storage_engine.put_content(_TEST_UPLOADS_PATH, _TEST_CONTENT)
+    assert storage_engine.exists(_TEST_UPLOADS_PATH)
+    assert storage_engine.get_content(_TEST_UPLOADS_PATH) == _TEST_CONTENT
+
+    # Test ensure fresh blobs are not deleted
+    storage_engine.clean_partial_uploads(timedelta(days=2))
+    assert storage_engine.exists(_TEST_UPLOADS_PATH)
+    assert storage_engine.get_content(_TEST_UPLOADS_PATH) == _TEST_CONTENT
+
+    # Test deletion of stale blobs
+    time.sleep(1)
```

Comment:
You could've also monkey patched the `LastModified` field from the response. But this should be fine for now.